src/main/java/com/ss/lms/controller/BorrowerController.java

```diff
@@ -1,187 +1,116 @@
-import org.springframework.boot.ApplicationArguments;
+
 import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
+
 
 import org.springframework.web.bind.annotation.RestController;
 
-import com.ss.lms.DAO.BookDAO;
-import com.ss.lms.DAO.BorrowerDAO;
-import com.ss.lms.DAO.BranchDAO;
-import com.ss.lms.DAO.CopiesDAO;
-import com.ss.lms.DAO.LoansDAO;
 import com.ss.lms.entity.Book;
 import com.ss.lms.entity.Borrower;
 import com.ss.lms.entity.Branch;
 import com.ss.lms.entity.Copies;
 import com.ss.lms.entity.Loans;
 import com.ss.lms.service.BorrowerService;
-import com.ss.lms.service.ConnectionUtil;
-
-
 
 @RestController
+@RequestMapping("borrower/")
 public class BorrowerController {
-	
+
 	@Autowired
 	BorrowerService borrowerService;
-	
-	@Autowired
-	Branch branch;
-	
-	
 
-	@RequestMapping(path = "borrower/{cardNo}/library/{branchId}/books/{bookId}:checkout", method = RequestMethod.POST, produces = {
-					MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE })
+
+	@PostMapping(path = "{cardNo}/branches/{branchId}/books/{bookId}:checkout", produces = {
+			MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE })
 	public ResponseEntity<Loans> checkOutBook(@PathVariable("cardNo") Integer cardNo,
-            @PathVariable("branchId") Integer branchId,
-            @PathVariable("bookId") Integer bookId) throws SQLException {
-		
-		Borrower borrower =null;
-		Book book = null;
-		Branch branch =null;
-		
-	
-		try {
-			
-			try {
-			
-			borrower = borrowerService.readByCardNoEssentialData (cardNo);
-			book = borrowerService.readBooksById(bookId);
-			branch = borrowerService.readBranchsById(branchId);
-			
-			Copies copy = borrowerService.readCopyByBranchIDBookID(book, branch);
-			Integer noOfCopies = copy.getNoOfCopies();
-			
-			if (noOfCopies == 0) {
-				return new ResponseEntity<Loans>(HttpStatus.NOT_FOUND);
-			}
-			}
-			catch (Exception e) {
-				return new ResponseEntity<Loans>(HttpStatus.NOT_FOUND);
-			}
-			
-			
-		}
-		catch (Exception e) {
-			return new ResponseEntity<Loans>(HttpStatus.INTERNAL_SERVER_ERROR);
+			@PathVariable("branchId") Integer branchId, @PathVariable("bookId") Integer bookId) {
+
+		Optional<Borrower> borrower = borrowerService.readBorrowerByCardNo(cardNo);
+		Optional<Book> book = borrowerService.readBooksById(bookId);
+		Optional<Branch> branch = borrowerService.readBranchsById(branchId);
+
+		if (!(borrower.isPresent() && book.isPresent() && branch.isPresent())) {
+			return new ResponseEntity<Loans>(HttpStatus.NOT_FOUND);
```

Comment:
I think the more efficent way would be to do all this in the service level itself. 
Then just call the .get() method which will throw a NoSuchElement exception that spring will handle for you